### PR TITLE
Add action to auto update gradle wrapper

### PR DESCRIPTION
**Pull request by jack-berg**

### CODE REVIEW

File: .github/workflows/update-gradle-wrappers-daily.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - id: setup-java
+        uses: actions/setup-java@v3
+        with:
+          distribution: temurin
+          java-version: 17
```
Comment (jack-berg) [inline .github/workflows/update-gradle-wrappers-daily.yml:20]: FYI @trask this task fails without installing the JDK. Didn't see any reference to this in the docs for the `update-gradle-wrapper-action`, but updating the wrapper requires running a gradle daemon, which requires java to be installed. Maybe `update-gradle-wrapper-action` installs a lower version of java than is required by opentelemetry-java?

Comment (trask) [reply .github/workflows/update-gradle-wrappers-daily.yml:20]: interesting, i'm guessing it didn't spit out a useful error to give us any clues?

**Review by trask — APPROVED** [review]

File: .github/workflows/update-gradle-wrappers-daily.yml (hunk 1 above)
Comment (jack-berg) [reply .github/workflows/update-gradle-wrappers-daily.yml:20]: Was testing this on my fork. Here's the build out put without setting up java: https://github.com/jack-berg/opentelemetry-java/actions/runs/5204819774/jobs/9389527088

File: .github/workflows/reusable-workflow-notification.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      - uses: actions/checkout@v3
+
+      - name: Open issue or add comment if issue already open
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          number=$(gh issue list --search "Workflow failed: $GITHUB_WORKFLOW" --limit 1 --json number -q .[].number)
+
+          echo $number
+          echo ${{ inputs.success }}
```
Comment (mateuszrzeszutek) [inline .github/workflows/reusable-workflow-notification.yml:25]: Nit: do we still need these debug prints?

**Review by mateuszrzeszutek — APPROVED** [review]

File: .github/workflows/reusable-workflow-notification.yml (hunk 2 above)
Comment (jack-berg) [reply .github/workflows/reusable-workflow-notification.yml:25]: Copied this from opentelemetry-java-instrumentation. I'm fine removing them if @trask thinks the script logic is reliable. I did test this in my personal fork and it seemed to work ok, but didn't test edge cases.

File: .github/workflows/update-gradle-wrappers-daily.yml (hunk 1 above)
Comment (trask) [reply .github/workflows/update-gradle-wrappers-daily.yml:20]: Ya I think we're good to remove

File: .github/workflows/reusable-workflow-notification.yml (hunk 2 above)
Comment (trask) [reply .github/workflows/reusable-workflow-notification.yml:25]: Ya I think we're good to remove

**Pull request merged**